### PR TITLE
Remove healthchecker ceph client

### PR DESCRIPTION
**Pull request by ezio-auditore**
Solves: https://github.com/red-hat-storage/ocs-operator/issues/2829

### CODE REVIEW

**Review by rewantsoni — CHANGES_REQUESTED** [review]

File: controllers/storageconsumer/storageconsumer_controller.go [hunk 1]
```diff
@@ -59,11 +59,10 @@ type StorageConsumerReconciler struct {
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 
-	ctx                     context.Context
-	storageConsumer         *ocsv1alpha1.StorageConsumer
-	cephClientHealthChecker *rookCephv1.CephClient
```
[the remark this answers is not in the document]
Comment (ezio-auditore) [reply controllers/storageconsumer/storageconsumer_controller.go:64]: Updated

**Comment by rewantsoni** [comment]
The changes look good, just wondering if we should update the name for setCephResourceStatus method and `CephResources` field in the StorageConsumer Status to reflect that they can be any kind of `Resources` (NoobaaAccount or any other fields we might want to add in future)

File: controllers/storageconsumer/consumer_test.go [hunk 2]
```diff
@@ -179,18 +147,13 @@ func TestCephName(t *testing.T) {
 				{
 					Kind:  "CephClient",
 					Name:  "rbd",
-					Phase: "Error",
+					Phase: "Ready",
```
Comment (Madhu-1) [inline controllers/storageconsumer/consumer_test.go:150]: we have only Noobaa in the status nothing else, right? why do we have ceph clients in the status field?

Comment (ezio-auditore) [reply controllers/storageconsumer/consumer_test.go]: Removed all cephclients

**Comment by ezio-auditore** [comment]
> The changes look good, just wondering if we should update the name for setCephResourceStatus method and `CephResources` field in the StorageConsumer Status to reflect that they can be any kind of `Resources` (NoobaaAccount or any other fields we might want to add in future)

@rewantsoni Yes I agree the name should be changed from CephResources to GenericResources or some other name to clearly justify the need and the purpose of the field.

**Review by Madhu-1 — COMMENTED** [review]
LGTM.

**Comment by Madhu-1** [comment]
> > The changes look good, just wondering if we should update the name for setCephResourceStatus method and `CephResources` field in the StorageConsumer Status to reflect that they can be any kind of `Resources` (NoobaaAccount or any other fields we might want to add in future)
> 
> @rewantsoni Yes I agree the name should be changed from CephResources to GenericResources or some other name to clearly justify the need and the purpose of the field.

This might need API change as API is tied to the CephResource name

**Review by iamniting — APPROVED** [review]

**Pull request merged**